src/lib/tls/tls_algos.h

```diff
@@ -117,48 +118,81 @@ enum class Group_Params : uint16_t {
-          group == Group_Params::FFDHE_6144 || group == Group_Params::FFDHE_8192;
-}
+      constexpr Group_Params(Group_Params_Code code) : m_code(code) {}
 
-constexpr bool is_pure_kyber(const Group_Params group) {
-   return group == Group_Params::KYBER_512_R3 || group == Group_Params::KYBER_768_R3 ||
-          group == Group_Params::KYBER_1024_R3;
-}
+      constexpr Group_Params(uint16_t code) : m_code(static_cast<Group_Params_Code>(code)) {}
 
-constexpr bool is_hybrid(const Group_Params group) {
-   return group == Group_Params::HYBRID_X25519_KYBER_512_R3_CLOUDFLARE ||
-          group == Group_Params::HYBRID_X25519_KYBER_768_R3_CLOUDFLARE ||
-          group == Group_Params::HYBRID_X25519_KYBER_512_R3_OQS ||
-          group == Group_Params::HYBRID_X25519_KYBER_768_R3_OQS ||
-          group == Group_Params::HYBRID_SECP256R1_KYBER_512_R3_OQS ||
-          group == Group_Params::HYBRID_SECP384R1_KYBER_768_R3_OQS ||
-          group == Group_Params::HYBRID_SECP521R1_KYBER_1024_R3_OQS;
-}
+      /**
+      * Returns std::nullopt if an unknown name
+      */
+      static std::optional<Group_Params> from_string(std::string_view group_name);
 
-constexpr bool is_kem(const Group_Params group) {
-   return is_pure_kyber(group) || is_hybrid(group);
-}
+      constexpr bool operator==(Group_Params_Code code) const { return m_code == code; }
 
-constexpr bool is_post_quantum(const Group_Params group) {
-   return is_pure_kyber(group) || is_hybrid(group);
-}
+      constexpr bool operator==(Group_Params other) const { return m_code == other.m_code; }
+
+      constexpr bool operator<(Group_Params other) const { return m_code < other.m_code; }
+
+      constexpr Group_Params_Code code() const { return m_code; }
+
+      constexpr uint16_t wire_code() const { return static_cast<uint16_t>(m_code); }
+
+      constexpr bool is_x25519() const { return m_code == Group_Params_Code::X25519; }
 
-std::string group_param_to_string(Group_Params group);
-Group_Params group_param_from_string(std::string_view group_name);
-std::vector<std::pair<std::string, std::string>> hybrid_group_param_to_algorithm_specs(Group_Params group);
-bool group_param_is_dh(Group_Params group);
+      constexpr bool is_ecdh_named_curve() const {
+         return m_code == Group_Params_Code::SECP256R1 || m_code == Group_Params_Code::SECP384R1 ||
+                m_code == Group_Params_Code::SECP521R1 || m_code == Group_Params_Code::BRAINPOOL256R1 ||
+                m_code == Group_Params_Code::BRAINPOOL384R1 || m_code == Group_Params_Code::BRAINPOOL512R1;
+      }
+
+      constexpr bool is_dh_named_group() const {
+         return m_code == Group_Params_Code::FFDHE_2048 || m_code == Group_Params_Code::FFDHE_3072 ||
+                m_code == Group_Params_Code::FFDHE_4096 || m_code == Group_Params_Code::FFDHE_6144 ||
+                m_code == Group_Params_Code::FFDHE_8192;
```

Comment:
I know this is somewhat redundant, but for clarity?

```suggestion
         if(!is_diffie_hellman()) {
            return false;
         }
         return m_code == Group_Params_Code::FFDHE_2048 || m_code == Group_Params_Code::FFDHE_3072 ||
                m_code == Group_Params_Code::FFDHE_4096 || m_code == Group_Params_Code::FFDHE_6144 ||
                m_code == Group_Params_Code::FFDHE_8192;
```

Reply:
No longer required since `is_diffie_hellman` is gone